### PR TITLE
Added benchmark, Moved minio-deps to fork without race fix we don't need.

### PR DESCRIPTION
**Pull request by bwplotka**
Fixes: https://github.com/thanos-io/thanos/issues/3917

Long term fix: https://github.com/thanos-io/thanos/issues/3967

Performed some benchmarks:

```
	// Uploading 100MB with:
	// PartSize: 1024 * 1024 * 100 = BenchmarkUpload-12    	      84	 416025119 ns/op	421129826 B/op	     946 allocs/op  (400MB allocated for single op)
	// ParSize:  1024 * 1024 * 25  = BenchmarkUpload-12    	     100	 273648888 ns/op	106639830 B/op	    1576 allocs/op (100MB overallocated).
	// PartSize: 1024 * 1024 * 8   = BenchmarkUpload-12    	      97	 254052232 ns/op	35392395 B/op	    3078 allocs/op (Still 33MB allocated [8x4])
```

I reverted the [race fix](https://github.com/minio/minio-go/pull/1357/files) locally on my branch and the result is: 

```
// PartSize: 1024 * 1024 * 100 = BenchmarkUpload-12    	      88	 282906636 ns/op	 1671249 B/op	     907 allocs/op
// PartSize: 1024 * 1024 * 8 =BenchmarkUpload-12    	     100	 300660587 ns/op	 2195440 B/op	    3044 allocs/op
```

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Review by kakkoyun — APPROVED** [review]
LGTM 🥇 
Can we add benchmark results for future reference?

**Review by prmsrswt — APPROVED** [review]
Looks good to me!

**Comment by bwplotka** [comment]
Done!

File: pkg/objstore/s3/s3_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+	"github.com/thanos-io/thanos/pkg/testutil"
+)
+
+// Regression benchmark for https://github.com/thanos-io/thanos/issues/3917.
+func BenchmarkUpload(b *testing.B) {
+	b.ReportAllocs()
+	ctx := context.Background()
+
+	s, err := e2e.NewScenario("e2e_bench_mino_client")
```
Comment (hitanshu-mehta) [inline pkg/objstore/s3/s3_e2e_test.go:26]: Typo :) `mino`

Comment (bwplotka) [reply pkg/objstore/s3/s3_e2e_test.go:26]: Will fix later (:

**Pull request merged**